module/zfs/zpl_file.c

```diff
@@ -361,8 +361,10 @@ zpl_write(struct file *filp, const char __user *buf, size_t len, loff_t *ppos)
 	ssize_t wrote;
 
 	crhold(cr);
+	spl_inode_lock(file_inode(filp));
 	wrote = zpl_write_common(filp->f_mapping->host, buf, len, ppos,
```

Comment:
Originally it was for backwards compatibility but since this code was originally written we've added a `file_inode()` compat wrapper to support all the way back to 2.6.32 kernels (which is the oldest kernel we support).  You _should_ be able to safely change this to `file_inode()`.